rfcs/2022-02-11-responsive-and-adaptive-layout-strategy.md

```diff
@@ -0,0 +1,403 @@
+- Ensuring that elements fit within a page is not enough. For a responsive design to be successful, the design must also be usable at all screen resolutions and sizes.
+  - An example of fluid search field which shrinks/grows based on screen width but isn't usable enough.
+    <img alt="Responsive Modal" src="./images/responsive-adaptive-layout/responsive-search-field.png">
+
+### Pros and Cons of Responsive Design
+
+| Pros                          | Cons                                                                                                                                                                                                                                                                                         |
+| ----------------------------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| Abundance of templates to use | Elements can move around fluidly                                                                                                                                                                                                                                                             |
+| Often easier to implement     | Longer mobile download times.                                                                                                                                                                                                                                                                |
+|                               | Tons of `@media` queries often leads to a maintenance overhead in code                                                                                                                                                                                                                       |
+|                               | Images and other high bandwidth features are simply scaled down rather than resized or eliminated outright(unless added some lines of JS which again degrades the performance and increases the mess in code), elements which may not be vital to user experience can cause poor performance |
+
+## Adaptive Design
+
+- While responsive design relies on changing the design pattern to fit the real estate available to it, **adaptive design** has multiple fixed layout sizes. When the site _detects_ the available space, it _selects_ the layout most appropriate for the screen. So, when you open a browser on the desktop, the site chooses the best layout for that screen, resizing the browser has no impact on the design(until it hit the breakpoint if we have mentioned it).
```

Comment:
> resizing the browser has no impact on the design(until it hit the breakpoint if we have mentioned it)

Didn't get this part. So, with this approach, we would download the appropriate bundle/styles when the system/server gets the request based on the initial screen height/width, right?
How would things change while resizing then even after a new breakpoint is hit?

Reply:
my bad. rephrasing this to

> a reload would be required to see the change